pyrmq/publisher.py

```diff
@@ -145,12 +145,18 @@ def connect(self, retry_count=1) -> (BlockingConnection, BlockingChannel):
             return self.connect(retry_count=(retry_count + 1))
 
     def publish(
-        self, data: dict, priority: Optional[int] = None, attempt=0, retry_count=1
+        self,
+        data: dict,
+        priority: Optional[int] = None,
+        message_properties: dict = None,
```

Comment:
Why not set `{}` here instead of `None` then setting it `{}` below?

Reply:
Also you should set it `Optional[dict]` because this field is not required

Reply:
Python dicts are mutable and should not be instantiated as method arguments. See [this](https://docs.python-guide.org/writing/gotchas/) for more info.